README.md

```diff
@@ -307,6 +306,28 @@ Unit tests are implemented in Jest and can be run via
 $ npm test
 ```
 
+### Typescript
+
+This library was first built using just JavaScript and transpiled using Babel.
+
+Migrating to Typescript has some clear advantages however and so since that time
+we've started a gradual migration to this being a Typescript project.
+
+When feasible, all future contributions should expand the use of Typescript as
+much as possible. `.js` files should be converted to `.ts` and errors fixed as
+necessary.
+
+In order to facilitate the gradual migration, babel is still used for transpiling
+while `tsc` is responsible for type checking and building declaration files. See
+[here](https://www.staging-typescript.org/docs/handbook/tutorials/babel-with-typescript.html#babel-for-transpiling-tsc-for-types) for more information.
+
+Due to some of the restrictions on the build process configs resulting from this decision,
+namely [isolatedModules](tsconfig.json), "global" types are handled in a `types/` directory
+as opposed to some other more common patterns.
+
+We might change the build process as possible to rely solely on `tsc` at which point some
+of the configurations as well as management of the type declarations can be altered.
+
```

Comment:
I understand the strategy here. 
Open question remains: "At which point does it become easier/more feasible to go full tsc?"